tools/create-yorkie-app/index.ts

```diff
@@ -0,0 +1,292 @@
+    | 'overwrite'
+    | 'packageName'
+    | 'framework'
+    | 'variant'
+  >;
+
+  try {
+    result = await prompts(
+      [
+        {
+          type: () => {
+            console.log(
+              '\n🔑 To run these examples, you need Yorkie API key.' +
+                '\nGet your API key at https://yorkie.dev/dashboard\n',
+            );
+
+            return 'text';
+          },
+          name: 'apiKey',
+          message: reset('API key:'),
+          initial: apiKeyMessage,
+          onState: ({ value }) => {
+            apiKey = value === apiKeyMessage ? '' : value;
+          },
+        },
+        {
+          type: argTargetDir ? null : 'text',
+          name: 'projectName',
+          message: reset('Project name:'),
+          initial: defaultTargetDir,
+          onState: (state) => {
+            targetDir = formatTargetDir(state.value) || defaultTargetDir;
+          },
+        },
+        {
+          type: () =>
+            !fs.existsSync(targetDir) || isEmpty(targetDir) ? null : 'confirm',
+          name: 'overwrite',
+          message: () =>
+            (targetDir === '.'
+              ? 'Current directory'
+              : `Target directory "${targetDir}"`) +
+            ` is not empty. Remove existing files and continue?`,
+        },
+        {
+          type: (_, { overwrite }: { overwrite?: boolean }) => {
+            if (overwrite === false) {
+              throw new Error(red('✖') + ' Operation cancelled');
+            }
+            return null;
+          },
+          name: 'overwriteChecker',
+        },
+        {
+          type: () => (isValidPackageName(getProjectName()) ? null : 'text'),
+          name: 'packageName',
+          message: reset('Package name:'),
+          initial: () => toValidPackageName(getProjectName()),
+          validate: (dir) =>
+            isValidPackageName(dir) || 'Invalid package.json name',
+        },
+        {
+          type:
+            argTemplate && TEMPLATES.includes(argTemplate) ? null : 'select',
+          name: 'framework',
+          message:
+            typeof argTemplate === 'string' && !TEMPLATES.includes(argTemplate)
+              ? reset(
+                  `"${argTemplate}" isn't a valid template. Please choose from below: `,
+                )
+              : reset('Select a framework:'),
+          initial: 0,
+          choices: FRAMEWORKS.map((framework) => {
+            const frameworkColor = framework.color;
+            return {
+              title: frameworkColor(framework.display || framework.name),
+              value: framework,
+            };
+          }),
+        },
+        {
+          type: (framework: Framework) =>
+            framework && framework.variants ? 'select' : null,
+          name: 'variant',
+          message: reset('Select a variant:'),
+          choices: (framework: Framework) =>
+            framework.variants.map((variant) => {
+              const variantColor = variant.color;
```

Comment:
Do variants always have the same color as the framework? If so, it would be nice to use `framework.color` instead of specifying the color in each variant.